lib/requeue_content.rb

```diff
@@ -0,0 +1,38 @@
+class RequeueContent
+  def initialize(number_of_items: nil)
+    @number_of_items = number_of_items
+  end
+
+  attr_accessor :number_of_items
+
+  def call
+    if number_of_items.present?
+      ContentItemFilter.filter(state: :published).limit(number_of_items).each do |content_item|
```

Comment:
Yep, that's pretty much it. 

Alternative might be to write some SQL to handle batching through a limited subset but I think that's overkill here - the limit is really only used when testing the rake task at this stage.